pci/src/vfio.rs

```diff
@@ -950,7 +957,12 @@ impl VfioCommon {
 
             let cap_next = self.vfio_wrapper.read_config_byte((cap_iter + 1).into())
                 & PCI_CONFIG_CAPABILITY_PTR_MASK;
-            if cap_next == 0 {
+
+            // Break out of the loop, if we either find the end or we have a broken device. This
+            // doesn't handle all cases where a device might send us in a loop here, but it
+            // handles case of a device returning 0xFF instead of implementing a real
+            // capabilities list.
+            if cap_next == 0 || cap_next == cap_iter {
```

Comment:
This commit makes sense to me that it handles a corner case where we stop break the loop if a cap pointer points to itself. But I don't see how returning `0xff` as the next cap pointer would be a problem, particularly with the previous commit where the last two bits are masked out. Basically, the loop would stop at the top level with condition `while cap_iter != 0`.

Reply:
Aha, I misread the code for `cap_next` and `cap_iter`.

Reply:
Thanks @snue for explaining. I just got back from vacation and saw that you already took care of this. :pray: